packages/at_onboarding_cli/lib/src/onboard/at_onboarding_service_impl.dart

```diff
@@ -321,15 +322,28 @@ class AtOnboardingServiceImpl implements AtOnboardingService {
     Duration retryInterval, {
     bool logProgress = true,
   }) async {
+    int retryAttempt = 1;
     while (true) {
       logger.info('Attempting pkam auth');
       if (logProgress) {
         stderr.write('Checking ... ');
       }
-      bool pkamAuthSucceeded = await _attemptPkamAuth(
-        atLookUp,
-        enrollmentIdFromServer,
-      );
+      bool pkamAuthSucceeded = false;
+      try {
+        pkamAuthSucceeded =
+            await _attemptPkamAuth(atLookUp, enrollmentIdFromServer);
+      } catch (e) {
```

Comment:
Does `_attemptPkamAuth` throw an exception if PKAM auth fails in a "normal" way (i.e. the enrollment has not yet been approved)?

Some code comments explaining what exactly is going on here would be useful

Reply:
OK - I'd suggest we now remove _attemptPkamAuth and move its code inline, as _attemptPkamAuth isn't used anywhere else